java/statement/builder/StatementTypeBuilder.java

```diff
@@ -46,72 +47,67 @@ default StatementType type(Graql.Token.Type type) {
         return type(type.toString());
     }
     
-    /**
-     * @param name a string that this variable's label must match
-     * @return this
-     */
     @CheckReturnValue
-    default StatementType type(String name) {
-        return type(new TypeProperty(name));
+    default StatementType type(String label) {
+        return type(new TypeProperty(Label.of(label, null)));
+    }
+
+    @CheckReturnValue
+    default StatementType type(String label, String scope) {
+        return type(new TypeProperty(Label.of(label, scope)));
+    }
+
+    @CheckReturnValue
+    default StatementType type(Label label) {
+        return type(new TypeProperty(label));
     }
 
-    /**
-     * set this concept type variable as abstract, meaning it cannot have direct instances
-     *
-     * @return this
-     */
     @CheckReturnValue
     default StatementType isAbstract() {
         return type(AbstractProperty.get());
     }
 
     default StatementType sub(Graql.Token.Type type) {
-        return sub(type.toString());
+        return sub(type.toString(), null);
+    }
+
+    @CheckReturnValue
+    default StatementType sub(String type, @Nullable String scope) {
```

Comment:
I think we don't need the annotation `@Nullable` here?

Reply:
right